filter.go

```diff
@@ -28,23 +28,27 @@ func New(n int, p float64, h func([]byte) (uint64, uint64)) Filter {
 	return &classicFilter{b: make([]byte, int(m/8)), k: int(k), h: h}
 }
 
+func (f *classicFilter) getOffset(x, y uint64, i int) uint64 {
+	return (x + uint64(i)*y) % (8 * uint64(len(f.b)))
+}
+
 func (f *classicFilter) Add(b []byte) {
 	x, y := f.h(b)
 	for i := 0; i < f.k; i++ {
-		offset := (x + uint64(i)*y) % (8 * uint64(len(f.b)))
+		offset := f.getOffset(x, y, i)
 		f.b[offset/8] |= 1 << (offset % 8)
 	}
 }
 
 func (f *classicFilter) Test(b []byte) bool {
 	x, y := f.h(b)
 	for i := 0; i < f.k; i++ {
-		offset := (x + uint64(i)*y) % (8 * uint64(len(f.b)))
+		offset := f.getOffset(x, y, i)
 		if f.b[offset/8]&1<<(offset%8) == 0 {
-			return false
+			return true
```

Comment:
Why reverse the truth? `Test` should return true if the item has been added before, and false otherwise.

Reply:
my fault. The correct  should be 

```
	for i := 0; i < f.k; i++ {
		offset := f.getOffset(x, y, i)
		if f.b[offset/8]&(1<<(offset%8)) == 0 {
			return false
		}
	}
	return true
```

but the FPR over 50%.